lib/class-factory.js

```diff
@@ -177,13 +182,36 @@ function ClassFactory (vm) {
           }
         };
 
-        C = ensureClass(getClassHandle, className);
+        try {
+          C = ensureClass(getClassHandle, className);
+        } finally {
+          setUsedClass(className, C);
+        }
       }
     }
 
     return new C(null);
   };
 
+  function getUsedClass(className) {
```

Comment:
```suggestion
  function getUsedClass (className) {
```

Could you run this code through [semistandard](https://github.com/standard/semistandard)? Noticed missing space after method names (like this), but there might be other minor issues needing fixing.